Sources/UIAlertControllerView.swift

```diff
@@ -69,22 +72,55 @@ class UIAlertControllerView: UIView {
             let previousElement = (index > 0) ? buttons[index - 1] : header
             button.sizeToFit()
             button.contentHorizontalAlignment = .left
-            button.bounds.size.height += verticalPadding
+            button.frame.size.height += verticalPadding
             button.frame.size.width = max(bounds.size.width, button.frame.size.width)
-            button.frame.origin = CGPoint(x: 0, y: previousElement.frame.maxY)
+            button.frame.origin.y = (previousElement == header)
+                ? previousElement.frame.maxY + verticalPadding
+                : previousElement.frame.maxY
+            button.titleLabelOriginX = horizontalPadding
+        })
+    }
+
+    private func layoutAsAlert() {
+        header.frame.origin = CGPoint(x: horizontalPadding, y: verticalPadding)
+        text.frame.origin.x = horizontalPadding
+        text.frame.width = UIScreen.main.bounds.width * 0.5
+        text.sizeToFit()
+        text.frame.minY = header.frame.maxY + verticalPadding
+        buttons.enumerated().forEach({ (arg) in
+            let (index, button) = arg
+            button.sizeToFit()
+            button.frame.width = max(button.frame.width, 75)
+            button.contentHorizontalAlignment = .center
+            button.frame.minY = text.frame.maxY + verticalPadding
+            if button == buttons.first {
+                button.frame.origin.x = bounds.maxX - button.frame.width - verticalPadding
+            } else {
+                let previousElement = buttons[index - 1]
+                button.frame.origin.x = previousElement.frame.minX - button.frame.width
```

Comment:
nit / optional:

```suggestion
                button.frame.origin.x = previousElement.frame.minX - button.frame.width - verticalPadding / 2
```

The buttons are right next to each other - perhaps some gap between them is better?

<img width="752" alt="Screenshot 2021-12-10 at 18 35 18" src="https://user-images.githubusercontent.com/11625008/145617361-f7b7f129-0ce2-4981-9807-9850994abd24.png">

<img width="752" alt="Screenshot 2021-12-10 at 18 36 04" src="https://user-images.githubusercontent.com/11625008/145617365-0c7a2559-2d5d-4e2f-b405-311515920598.png">

Reply:
BTW - we only layout the buttons horizontally.
Hopefully for our current use case this works fine.
On Android usually when the width of the buttons get larger the buttons[ layout switches to vertical](https://material.io/components/dialogs#usage):

<img src="https://lh3.googleusercontent.com/QLDP9YlL2JDE2yAD8fdbDbCnnkvFfoaJqMVmd30k2Gq1QQs4PyNmlLG0LMZugq238LofhjiQ50fXHaqRY8svC-4OaXMNjS-3MTqYeUQ=w1064-v0" />